frontend/app/routes/manga/index.js

```diff
@@ -1,12 +1,86 @@
 import Ember from 'ember';
 import setTitle from '../../utils/set-title';
+/* global Messenger */
 
 export default Ember.Route.extend({
   model: function() {
     return this.modelFor('manga');
   },
 
+  setupController: function(controller, model) {
+    controller.set('model', model);
+  },
+
   afterModel: function(resolvedModel) {
     return setTitle(resolvedModel.get('romajiTitle'));
+  },
+
+  saveMangaLibraryEntry: function(mangaLibraryEntry) {
+    var manga = this.currentModel;
+    return Messenger().expectPromise((function() {
+      return mangaLibraryEntry.save();
+    }), {
+      progressMessage: "Saving " + manga.get('romajiTitle') + "...",
+      successMessage: function() {
+        // update the 'full-manga' relationship since it seems
+        // to disappear into the void
+        manga.set('mangaLibraryEntry', mangaLibraryEntry);
+        return "Saved " + manga.get('romajiTitle') + "!";
+      }
+    });
+  },
+
+  actions: {
+    loadFullCharacters: function() {
+      var manga_id = this.currentModel.get('id');
+      this.store.find('casting', { manga_id }).then((castings) => {
+        this.controller.set('fullCharacters', castings);
+      });
+    },
+
+    toggleFavorite: function() {
+      if (!this.get('currentUser.isSignedIn')) {
+        alert('Need to be signed in');
```

Comment:
We actually have it wrapped in `{{#if libraryEntryExists}}` so I'm not sure why the action checks again